ext/standard/file.c

```diff
@@ -945,9 +945,9 @@ PHP_FUNCTION(popen)
 
 	/* Musl only partially validates the mode. Manually check it to ensure consistent behavior. */
 	if (mode_len != 1 || (*posix_mode != 'r' && *posix_mode != 'w')) {
-		php_error_docref2(NULL, command, posix_mode, E_WARNING, "Invalid mode");
+		zend_argument_value_error(2, "must be either \"r\" or \"w\"");
 		efree(posix_mode);
-		RETURN_FALSE;
+		RETURN_THROWS();
```

Comment:
As this now throws ValueError, we should make it consistent between Linux & Windows. I.e. move this outside the ifndef, but also allow `b` (I think we may require that first char is r/w and only allow b as second).

Reply:
Hopefully I managed to find the condition you imagined ^^